packages/xo-server/src/api/custom-field.js

```
@@ -0,0 +1,52 @@
+const KEY = 'XenCenter.CustomFields.'
+
+export async function add({ object, name, value }) {
+  await this.getXapiObject(object).update_other_config(KEY + name, value)
+}
+
+add.description = 'Add a new costum field to an object'
+
+add.params = {
+  id: { type: 'string' },
+  name: { type: 'string' },
+  value: { type: 'string' },
+}
+
+add.resolve = {
+  object: ['id', null, 'administrate'],
+}
+
+// -------------------------------------------------------------------
+
+export async function remove({ object, name }) {
+  await this.getXapiObject(object).update_other_config(KEY + name, null)
+}
+
+remove.description = 'Remove an existing custom field from an object'
+
+remove.params = {
+  id: { type: 'string' },
+  name: { type: 'string' },
+}
+
+remove.resolve = {
+  object: ['id', null, 'administrate'],
+}
+
+// -------------------------------------------------------------------
+
+export async function set({ object, name, value }) {
```

It's like `add` function but I think it's more clear for the client side that we `set` the custom field.

---

A `set` in the most APIs is `add + update`. And `add` should throw an error if an entry exists.

---

https://github.com/vatesfr/xen-orchestra/blob/7a8ca2f0689811045936f1f5653a2c52dc34a2fe/packages/xen-api/src/index.js#L311-L326

If the entry exists and there is no error: add + delete + add (update)
IMHO, it's not a bad idea to use only the `update_other_config`  for `add` and `set` function.
What do you think?

---

By definition, a set shouldn't act like an add. 

An add shouldn't overwrite an existing entry.